File: apps/Lodgr.Api/Models/Ward.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Lodgr.Api.Models;

[Table("wards")]
public class Ward
{
    [Key]
    [StringLength(20)]
    [Column("code")]
    public string Code { get; set; } = string.Empty;

    [Required]
    [StringLength(20)]
    [Column("district_code")]
    public string DistrictCode { get; set; } = string.Empty;

    [Required]
    [StringLength(100)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(150)]
    [Column("full_name")]
    public string FullName { get; set; } = string.Empty;

    [StringLength(50)]
    [Column("administrative_unit")]
    public string? AdministrativeUnit { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}

File: apps/api/src/main/java/io/github/hoanghonghuy/lodgr/entity/address/Ward.java
package io.github.hoanghonghuy.lodgr.entity.address;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "wards")
@Getter
@Setter
public class Ward {
    @Id
    @Column(name = "code", length = 10, nullable = false)
    private String code;

    @Column(name = "code_name", length = 50)
    private String codeName;

    @Column(name = "code_name_en", length = 50)
    private String codeNameEn;

    @Column(name = "name", length = 100)
    private String name;

    @Column(name = "name_en", length = 100)
    private String nameEn;

    @ManyToOne
    @JoinColumn(name = "district_code")
    private District district;
}
